### PR TITLE
Add blue print data providers to minecolonies

### PR DESCRIPTION
**Pull request by someaddons**
# Changes proposed in this pull request:
Allows building and decoration TE's to hold special blueprint TE data

Fix tavern autohire
Fix raids not always obeying the min nights
Fix player login loading chunks during network login event
Improve guard experience gain

Review please

### CODE REVIEW

File: src/api/java/com/minecolonies/api/tileentities/AbstractTileEntityColonyBuilding.java [hunk 1]
```diff
@@ -142,4 +148,77 @@ public static boolean isInTileEntity(final ICapabilityProvider entity, @NotNull
...
+    public void setPositionedTags(final Map<BlockPos, List<String>> positionedTags)
+    {
+        tagPosMap = positionedTags;
+    }
+
+    /**
+     * Corner positions of schematic, relative to te pos.
+     */
+    private BlockPos corner1 = BlockPos.ZERO;
+    private BlockPos corner2 = BlockPos.ZERO;
```
Comment (Raycoms) [inline src/api/java/com/minecolonies/api/tileentities/AbstractTileEntityColonyBuilding.java]: Please put those over all the methods.

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingTavern.java [hunk 2]
```diff
@@ -99,6 +99,7 @@ public void onPlayerEnterBuilding(final PlayerEntity player)
     @Override
     public void onColonyTick(@NotNull final IColony colony)
     {
+        super.onColonyTick(colony);
         if (musicCooldown > 0)
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingTavern.java:103]: really, that was the problem?

**Comment by Raycoms** [comment]
Only one small comment, else :shipit: 

[Approved with PullApprove](https://pullapprove.com/ldtteam/minecolonies/pull-request/5021/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingTavern.java (hunk 2 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingTavern.java:103]: ye forgot the super call which has the autohiring logic :D

**Review by Asherslab — APPROVED** [review]
Looks good. 
Decoration controller working any better with this?

**Comment by someaddons** [comment]
deco controller only gets the corner positions additionally I think, its overlapping a bit with the existing functionality of it

**Pull request merged**